lib/workers/repository/update/branch/reuse.ts

```diff
@@ -34,21 +34,44 @@ async function shouldKeepUpdated(
 
 export async function shouldReuseExistingBranch(
   config: BranchConfig,
-): Promise<ParentBranch> {
+): Promise<BranchConfig> {
   const { baseBranch, branchName } = config;
+  // eslint-disable-next-line no-param-reassign
+  config = { ...config };
   const result: ParentBranch = { reuseExistingBranch: false };
```

Comment:
```suggestion
   const result: BranchConfig = { ...config, reuseExistingBranch: false };
```
then return result again, so less changes.

Reply:
It needs a bit more than that: Since we mutate the `rebaseWhen` property of the config, we need to perform this mutation on `result` and not on `config` after this config merge because the change it wouldn't be part of the returned `result` object otherwise. And for consistency, all further references to the config in this function should use `result` because that's the up-to-date config object. But the failing tests from before are still failing and I can't think of an explanation.

See 2831d2f2e632d5384f5c75c81789a1805ea2eb2d.

Reply:
@viceice was this resolved?

Reply:
yes